src/components/data/Scroll.js

```diff
@@ -0,0 +1,44 @@
+import classnames from 'classnames';
+
+import { downcastRef } from '../../util/typing';
+
+/**
+ * @typedef {import('../../types').PresentationalProps} CommonProps
+ * @typedef {import('preact').JSX.HTMLAttributes<HTMLElement>} HTMLAttributes
+ *
+ * @typedef ScrollProps
+ * @prop {'raised'|'flat'} [variant='raised'] - Render with scroll-hinting
+ *   shadows ('raised') or without ('flat')
+ */
+
+/**
+ * Render a fluid container that scrolls on overflow.
+ *
+ * @param {CommonProps & ScrollProps & HTMLAttributes} props
+ */
+export default function Scroll({
+  children,
+  classes,
+  elementRef,
+
+  variant = 'raised',
+
+  ...htmlAttributes
+}) {
+  return (
+    <div
+      {...htmlAttributes}
+      ref={downcastRef(elementRef)}
+      className={classnames(
+        // Prevent vertical flex or grid blowout: keep this contained vertically
+        // to parent by setting a min width
```

Comment:
```suggestion
        // to parent by setting a min height
```

I would also be fine with delegating most of the details to the Stack Overflow answer you mentioned (eg. ``// Prevent overflow by overriding `min-height: auto`. See https://stackoverflow.com/a/66689926/434243.``)